src/osemgrep/cli_scan/Scan_subcommand.mli

```diff
@@ -26,16 +26,17 @@ val run_scan_conf : Scan_CLI.conf -> Exit_code.t
    `diff_targets` are automatically calculated based on the output
    from the `git diff` command between the baseline commit and the
    head commit. *)
-type diff_config = { diff_targets : Fpath.t list; diff_depth : int option }
-
-val default_diff_config : diff_config
+type diff_config = {
+  diff_targets : Fpath.t list option;
```

Comment:
Can you move the comment in the PR about the different situation in the code here too?
Can you explain the different scenario and what are the values for each field in those different situation?
Also it looks like for --baseline-commit, the depth is always 0 and diff_targets represent the whole set of files to analyze whereas in pro depth will be different than 0 (which value actually? where this is set?) and maybe diff_targets are starting point to find the final set of targets? In that case maybe it would be better to introduce a better single type instead of those 2 fields like | BaseLine of Fpath.t list | Depth of Fpath.t list (* starting points *) * int (* depth *) | ...

Reply:
Thank you for the helpful suggestion. I've updated the type of the diff configuration, and it now conveys clearer semantics. It looks like the code has much better readability.